File: selfstudy/python/stars_sortdesc.py
'''
task 190:
Stars - SortDesc
You need to sort the values in descending order for star luminosities
'''

def sort_down_stars(brightness):
    return sorted(brightness, reverse=True)

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], sort_down_stars([10, 20, 30])),
        ([50, 60, 70], sort_down_stars([50, 60, 70])),
        ([1, 2, 3], sort_down_stars([1, 2, 3])),
        ([100, 200], sort_down_stars([100, 200])),
        ([], sort_down_stars([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = sort_down_stars(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# done

File: selfstudy/js/stars_sortdesc.js
/*
task 90:
Stars - SortDesc
You need to sort the values in descending order for star luminosities
*/

function sort_down_stars(brightness) {
  return [...brightness].sort((a, b) => b - a);
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: sort_down_stars([10, 20, 30]) },
  { input: [50, 60, 70], expected: sort_down_stars([50, 60, 70]) },
  { input: [1, 2, 3], expected: sort_down_stars([1, 2, 3]) },
  { input: [100, 200], expected: sort_down_stars([100, 200]) },
  { input: [], expected: sort_down_stars([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('sort_down_stars', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(sort_down_stars(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, sort_down_stars(t.input) === t.expected || JSON.stringify(sort_down_stars(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// done
